tasks/partial_bundle.js

```diff
@@ -7,39 +8,97 @@ var common = require('./util/common');
 var _bundle = require('./util/browserify_wrapper');
 
 var header = constants.licenseDist + '\n';
+var allTransforms = constants.allTransforms;
 var allTraces = constants.allTraces;
 var mainIndex = constants.mainIndex;
 
-var argv = process.argv;
+function isFalse(a) {
+    return (
+        a === 'none' ||
+        a === 'false'
+    );
+}
+
+function inputBoolean(a, dflt) {
+    return !a ? dflt : !isFalse(a);
+}
+
+function inputArray(a, dflt) {
+    dflt = dflt.slice();
+
+    return (
+        isFalse(a) ? [] :
+            !a || a === 'all' ? dflt :
+                a.split(',')
+    );
+}
 
-if(argv.length > 2) {
+if(process.argv.length > 2) {
     // command line
 
-    var traceList = ['scatter']; // added by default
-    var name;
-    for(var i = 2; i < argv.length; i++) {
-        var a = argv[i];
+    var args = minimist(process.argv.slice(2), {});
+
+    // parse arguments
+    var out = args.out ? args.out : 'custom';
+    var traces = inputArray(args.traces, allTraces);
+    var transforms = inputArray(args.transforms, allTransforms);
+    var calendars = inputBoolean(args.calendars, true);
```

Comment:
presumably at some point we will make more of the [components registered in core](https://github.com/plotly/plotly.js/blob/d0ddf03850ccfc5e625da0735b6dcea8e1d61a05/src/core.js#L32-L48) optional. In order to support this in a backward-compatible way, perhaps we can make an option like `skipcomponents=calendars`, to later expand to allow skipping other components?

Reply:
I also thought about this a bit. But I think another option that we want to provide in future is which calendar(s) to add. That can explain why `calendars` logic here is quite similar to `transforms` and `traces`. For the moment it is handled by a boolean; but it could be expanded to a list.
Perhaps to avoid confusion we can move `calendars` code from `./components/calendars` to `./calendars` now or once we split it?

Reply:
As discussed on slack, for now let's drop the calendars option - always include calendars in every bundle - until we can ensure that unrelated functionality like `connectgaps` will not break as in https://github.com/plotly/plotly.js/issues/4883.

At ~81kB minified and rarely used, it would be very nice to make these optional eventually, and you're right to want flexibility even when world calendars are requested to only include the necessary ones, not [all of them](https://github.com/plotly/plotly.js/blob/master/src/components/calendars/calendars.js)